# HG changeset patch
# User Cole Robinson <crobinso@redhat.com>
# Date 1236630113 14400
# Node ID 6126a50801deafa155b17b66cfc3008ffe584b24
# Parent  3f37d0519b1750666e88e7eaef8d5dfaafc34237
Fix conn uri lookup for domain migration.

diff -rup virt-manager-0.6.1/src/virtManager/domain.py new/src/virtManager/domain.py
--- virt-manager-0.6.1/src/virtManager/domain.py	2009-09-13 20:25:12.000000000 -0400
+++ new/src/virtManager/domain.py	2009-09-13 20:24:29.000000000 -0400
@@ -1293,11 +1293,15 @@ class vmmDomain(gobject.GObject):
         else:
             self._disk_io = self._sample_disk_io_dummy
 
-
-    def migrate(self, dictcon):
+    def migrate(self, destconn):
         flags = 0
         if self.lastStatus == libvirt.VIR_DOMAIN_RUNNING:
             flags = libvirt.VIR_MIGRATE_LIVE
-        self.vm.migrate(self.connection.vmm, flags, None, dictcon.get_short_hostname(), 0)
+
+        newxml = self.get_xml()
+
+        self.vm.migrate(destconn.vmm, flags, None, None, 0)
+
+        destconn.define_domain(newxml)
 
 gobject.type_register(vmmDomain)
